File: eco-commerce-front/src/app/Models/produit.ts
import { imageProduit } from "./imageProduit";

export class produit {

    id:number;
	nom:string;
	description:string;
	image:string;
    prix:number;
    imagePrd:imageProduit;


    constructor(id?:number,nom?:string,description?:string,image?:string,prix?:number){
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.image = image;
        this.prix = prix;
    }

}

File: eco-commerce/src/main/java/com/ecocommerce/Entity/Produit.java
package com.ecocommerce.Entity;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

//classe correspondant à la table produit dans la base de données
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Produit {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE,generator = "produit_seq")
	private Long id;
	private String nom;
	private String description;
	private String image;
	private Double prix;

}
